sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/query/search/resolver/impl/AbstractFiltersResolver.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in comHpliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.stargate.sgv2.docsapi.service.query.search.resolver.impl;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import io.stargate.sgv2.docsapi.service.common.model.Paginator;
+import io.stargate.sgv2.docsapi.service.query.executor.QueryExecutor;
+import io.stargate.sgv2.docsapi.service.query.model.RawDocument;
+import io.stargate.sgv2.docsapi.service.query.search.resolver.DocumentsResolver;
+import io.stargate.sgv2.docsapi.service.query.search.resolver.filter.CandidatesFilter;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+
+public abstract class AbstractFiltersResolver implements DocumentsResolver {
+
+  protected abstract DocumentsResolver getCandidatesResolver();
+
+  protected abstract Collection<CandidatesFilter> getCandidatesFilters();
+
+  protected abstract Multi<RawDocument> resolveSources(
+      RawDocument rawDocument, List<Uni<Boolean>> sources);
```

Comment:
* document
* resolve source must emit only one item that matches the document (maybe add test)